src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class SettingsStorageHelper
+{
+public:
+    SettingsStorageHelper(const QString &keyName)
+        : m_keyName(keyName)
+    {
+    }
+
+    void set(const T &value)
+    {
+        SettingsStorage::instance()->storeValue(m_keyName, value);
+    }
+
+    T get(const QVariant &defaultValue = QVariant())
+    {
+        return SettingsStorage::instance()->loadValue(m_keyName, defaultValue).template value<T>();
+    }
+
+private:
+    QString m_keyName;
```

Comment:
> MO, it's overkill.

For me, overkill is the storing ASCII text in UTF-16, overkill is the usege of `QVariant` (two words) when in the overwhelming majority of cases the value is either `bool` or a number. A bit of additional work for compiler is not an overkill.